src/components/RadioGroup/RadioGroup.tsx

```diff
@@ -0,0 +1,186 @@
+
+  static create: Function
+
+  static propTypes = {
+    /** Accessibility behavior if overridden by the user. */
+    accessibility: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Value of the currently checked radio item. */
+    checkedValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Initial checkedValue value. */
+    defaultCheckedValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+
+    /** Shorthand array of props for RadioGroup. */
+    items: customPropTypes.collectionShorthand,
+
+    /**
+     * Called after radio group value is changed.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All value props.
+     */
+    checkedValueChanged: PropTypes.func,
+
+    /** Custom styles to be applied for component. */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Custom variables to be applied for component. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** A vertical radio group displays elements vertically. */
+    vertical: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    as: 'div',
+    accessibility: RadioGroupBehavior as Accessibility,
+  }
+
+  static handledProps = [
+    'accessibility',
+    'as',
+    'checkedValue',
+    'checkedValueChanged',
+    'children',
+    'className',
+    'defaultCheckedValue',
+    'items',
+    'styles',
+    'variables',
+    'vertical',
+  ]
+
+  static autoControlledProps = ['checkedValue']
+
+  static Item = RadioGroupItem
+
+  actionHandlers: AccessibilityActionHandlers = {
+    nextItem: event => this.setCheckedItem(event, 1),
+    prevItem: event => this.setCheckedItem(event, -1),
+  }
+
+  setCheckedItem = (event, direction) => {
+    const nextItem = this.findNextEnabledCheckedItem(direction)
+
+    if (nextItem) {
+      this.trySetState({ checkedValue: nextItem.value })
+      _.invoke(this.props, 'checkedValueChanged', event, nextItem)
+    }
+
+    event.preventDefault()
+  }
+
+  findNextEnabledCheckedItem = (direction): IRadioGroupItemProps => {
+    if (!this.props.items || this.props.items.length === 0) {
+      return undefined
+    }
+
+    const currentIndex = this.props.items.findIndex(
+      item => (item as IRadioGroupItemProps).value === this.state.checkedValue,
+    )
+
+    for (
+      let newIndex = currentIndex + direction;
+      newIndex !== currentIndex;
+      newIndex += direction
+    ) {
+      if (newIndex < 0) {
+        newIndex = this.props.items.length - 1
+      } else if (newIndex >= this.props.items.length) {
+        newIndex = 0
+      }
+
+      if (newIndex < 0 || newIndex >= this.props.items.length || newIndex === currentIndex) {
```

Comment:
```newIndex < 0 || newIndex >= this.props.items.length``` you have already checked it above, I think ```newIndex === currentIndex``` might be enough

Reply:
yes, it's checked twice, but it is changed in between these two ifs.

Reply:
Sorry, I might be confused or blind, but as I see, you change it in between these two ifs to either last item index or first item index, which seems to be safe and no need to check it later again